src/Compilers/Core/Portable/Operations/ITupleExpression.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Represents a tuple expression.
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.
+    /// </remarks>
+    public interface ITupleExpression : IOperation
+    {
+        /// <summary>
+        /// Elements for tuple expression.
+        /// </summary>
+        ImmutableArray<IOperation> Elements { get; }
```

Comment:
It wsa an early decision we consciously made.  There are negatives to the materialization approach, but it puts us in a good space in terms of defining the tree and getting a good test infrastructure in place for it.  We can then optimize it later as necessary as an implementation detail.